### DISCUSSION TITLE
Support for oven

### DISCUSSION THREAD

**Discussion by Altar82** [ofalvai/home-assistant-candy#41] [Device support improvements]
hi,i have oven. You can add it on custom_component? If you want i can try some testing. tell me what i have to do.
For now i get this informations:

simplyfi 192.168.188.51 key read
{
        "statusForno":{
                "StatoWiFi":"0",
                "CodiceErrore":"E0",
                "RecipeId":"NULL",
                "RecipeStep":"0",
                "StartStop":"1",
                "Pausa":"0",
                "SicurezzaBambini":"0",
                "Selettore":"1",
                "Program":"2",
                "TempSet":"0",
                "TempRead":"310",
                "TempSetRaggiunta":"0",
                "DelayStart":"0",
                "RemainingTimeProgr":"65535",
                "ora":"9",
                "min":"2",
                "sec":"5",
                "TimeProgr":"0",
                "FWver":"004B",
                "ts":"0"
        }
}

Anyway i tried to integrate it but i have this error:

Error communicating with API: Exception('Unable to detect machine type from API response', {'statusForno': {'StatoWiFi': '0',

Ty so much.

**Comment by ofalvai** [comment]
Hello @Altar82, I can add some very basic support for ovens based on the API response you posted. But please help me understand what each value means and how to convert them to meaningful Home Assistant attributes. For reference, here is how washing machines and tumble dryers are created from the API response: https://github.com/ofalvai/home-assistant-candy/blob/main/custom_components/candy/client/model.py#L87

**Reply by voodooo82** [reply]
i can traduce from italian to english this state:

"statusForno":{  Status oven
"StatoWiFi":"0", status wifi
"CodiceErrore":"E0", error code
"RecipeId":"NULL", i don't know
"RecipeStep":"0", i don't know
"StartStop":"1 ", start stop
"Pausa":"0", pause
"SicurezzaBambini":"0", children secure
"Selettore":"1",  i don't know
"Programma":"2", cooking program
"TempSet":"0",  temperature setting
"TempRead":"310",  temperature read
"TempSetRaggiunta":"0",  temperature setting is reached
"DelayStart":"0",
"RemainingTimeProgr":"65535", is simple :)
"ora":"9", hour
"min":"2", minute
"sec":"5", second 
"TimeProgr":"0", time for program
"FWver":"004B", firmware version?
"ts":"0" ....

**Comment by Altar82** [comment]
sure, just give me a few days to do some testing (when my wife will be  not at home 😂😂). I will write here. Thank you very much.

**Comment by Altar82** [comment]
hello @ofalvai ,i get some information from my oven. tell me if is enough

https://pastebin.com/eyGbNjq4

tell me too if i can do some others test

ty so much

**Comment by ofalvai** [comment]
Hey @Altar82, thank you so much for the detailed API responses.

I added some initial oven support and released it as `0.3.0`, please let me know how it works in the real world.

I decided to skip mapping the program numbers to program names because I'm afraid these are not consistent between oven models. At least this is my experience with washing machine programs 😩

**Comment by Altar82** [comment]
Hi @ofalvai Ty so much for your work. i tried to install it and it created 2 sensors. Sensor.oven (I've seen some attributes, so I'll make templates to know what program name it is in). Sensor.oven_temperature, on the other hand, shows me 171 ° C but the oven is off and I used it yesterday as the last time. I will try to measure the temperature with a thermometer, to understand well how it measures the temperature. I'll do more tests and update you. Thanks so much again

**Comment by ofalvai** [comment]
The weird temperature value is probably the last reading from the previous session. My washing machine does something similar with spin speed and temperature while idle and only updates the values once the new program starts.

**Comment by mirkochip88** [comment]
Hi @Altar82 , how you get the key from oven? The tool give me "error: find_xor_key_list, could not find key". The same error you post on tool repository. Thanks.

**Comment by Altar82** [comment]
> Hi @Altar82 , how you get the key from oven? The tool give me "error: find_xor_key_list, could not find key". The same error you post on tool repository. Thanks.
Hello @mircochip88 ,i got Key with this tool on Linux,searching 
' "StatoWiFi":"1",',if i Remember well
https://community.home-assistant.io/t/dishwasher-candy-simply-fi-cdi-6015-wifi/136543/41

**Comment by Paddel0815** [comment]
I have the same Problem with the temperature. 

The temperature may have been multiplied by 5.
I set the oven to 210 degrees Celsius and the oven outputs 1149 degrees Celsius.
